### DISCUSSION TITLE
Issues authenticating with Twitter login "error_description=Error+getting+user+email+from+external+provider"

### DISCUSSION THREAD

**Discussion by birdiewd** [supabase/supabase#4677] [Questions] [answered]
I am having issues using Twitter authentication with my NextJS application. It is very bare-bones, authentication is the only thing I'm trying to tackle at the moment. I have a button that triggers a call to supabase. I see the Twitter authorization screen (I sign in using my twitter account). The application shows up in my Twitter profile as an allowed app. But my user information is not stored in the supabase user table. I only have Twitter login enabled, nothing else. All ids/keys/secrets have been triple checked and there are no leading/trailing spaces, and URLs matching as official documentation prescribes.

Thank you to any that help and I hope this helps others if there is something silly I am overlooking.

My **app**, if anybody wants to see it for themselves:
https://twitter-likes.vercel.app

**Screenshots** if you don't want to click the link:

![LLT5HFPJSV](https://user-images.githubusercontent.com/6692954/147416233-22cdb43f-395d-4754-a0ac-84a87ca31d18.png)
![b9hbJayQDW](https://user-images.githubusercontent.com/6692954/147416237-35b52331-233b-4e3b-baca-51b578233b44.png)
![oHyva69Mj0](https://user-images.githubusercontent.com/6692954/147416240-d75f0edb-a36a-4643-805e-3b7337c833a4.png)
![A1n24vK3xb](https://user-images.githubusercontent.com/6692954/147416243-34edddfb-d794-4f37-937b-366296a5b293.png)

Very simple **code** implementation:

![Code_oMgCQpUSNq](https://user-images.githubusercontent.com/6692954/147416246-6b145be5-dc44-4776-b4e7-80d2ddd23c1d.png)

**Comment by birdiewd** [comment, marked as answer]
This was my fault. I was doing all of this with an "Essentials" account instead of an "Elevated" account. At this time, supabase uses v1 of the Twitter API which does not allow access at the "Essentials" level.

**Reply by jonathangiardino** [reply]
Thanks, man, this comment saved me from a really frustrating afternoon 😅

**Comment by samjmck** [comment]
I am also getting this error. Not sure if it has with "Essentials" or "Elevated" account, I can't find anything about that in either the Twitter or Supabase docs. However, the option for "3-legged OAuth" as explained in [these Supabase docs](https://supabase.com/docs/guides/auth/auth-twitter) seems to be outdated.

**Reply by samjmck** [reply]
Related: https://github.com/supabase/supabase/issues/8127#issuecomment-1218780476

**Reply by ethanchiasson** [reply]
getting this issue even after having an "Elevated account". Do not believe it is a setup issue on my part since I am getting a 200 status error and tried all other varieties of API key pairs since the docs are so outdated. Not sure what is going on, since some people claim their apps work after receiving elevated status, but mine isn't.
